woocommerce/woocommerce
# Set default weight and dimension units based on country

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/53651

We defaults to metric units (kg and cm) in the Settings > Products page, which can lead to a poor user experience for U.S.-based users who primarily use imperial units (lbs and inches).

This PR makes the following changes:

1. Ensure U.S. installations default to imperial units
2. Automatically updates the weight and dimension units in product settings when the home country is updated via the Core Profiler.
3. Corrected weight and dimension units for specific countries in locale-info.php.
   - The US primarily uses pounds (lbs) for weight and inches (in) for dimensions.
   - US territories like Guam (GU), American Samoa (AS), US Virgin Islands (VI) also use imperial units.
   - UK [officially uses metric](https://www.gov.uk/weights-measures-and-packaging-the-law), though many retailers use imperial units. 
4. Update `core-profiler.md`
5. Update tests to reflect the changes



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Review (use AI tool) the locale-info.php file for the weight and dimension units for each country and make sure the units are set correctly. 

**Test new stores setup:**

1. Create a fresh site
2. Go to `/wp-admin/admin.php?page=wc-settings&tab=products`
3. Confirm that `Weight unit` and `Dimension unit` are set to `lbs` and `in` respectively.
4. Go to Core Profiler and choose `Set up my store`
6. Update the country to `United Kingdom` on about store step.
7. Confirm that `Weight unit` and `Dimension unit` are set to `kg` and `cm` respectively.
8. Go to Core Profiler and choose `Skip guided setup`
9. Select `Guam` as the country
10. Confirm that `Weight unit` and `Dimension unit` are set to `lbs` and `in` respectively.

Optional:

- Repeat the steps above for other US territories like American Samoa (AS), US Virgin Islands (VI) and confirm that the units are set to `lbs` and `in` respectively.

**Existing stores should not be affected**

1. Create a fresh site with latest stable WooCommerce and beta tester
2. Skip or complete core profiler setup or disable core profiler
3. Check the weight and dimension units in the settings page
4. Update WooCommerce to include this PR (using live branch feature)
5. Confirm that `Weight unit` and `Dimension unit` are remain the same as before the update.



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
